### DISCUSSION TITLE
How to add new TTS voices in koboldcpp？

### DISCUSSION THREAD

**Discussion by Ranger013** [LostRuins/koboldcpp#709] [Q&A]
Hello, today I was using the TTS function in koboldcpp, but I found that the built-in TTS voices were not very effective. I wanna add some new voices, how can i get this? Thank you!
<img width="380" alt="屏幕截图 2024-02-24 185906" src="https://github.com/LostRuins/koboldcpp/assets/32187038/786d314f-977e-4746-8d52-757f0fea13ea">

**Comment by LostRuins** [comment]
You need to setup XTTS API server. You can use the google colab version here: https://colab.research.google.com/drive/1b-X3q5miwYLVMuiH_T73odMO8cbtICEY?usp=sharing

Connect with the tunnel URL.

**Comment by illtellyoulater** [comment]
I made a PR to enable support for https://github.com/erew123/alltalk_tts

https://github.com/LostRuins/koboldcpp/pull/719
